test/gtest/ucp/test_ucp_sockaddr.cc

```diff
@@ -723,6 +733,85 @@ UCS_TEST_P(test_ucp_sockaddr, err_handle) {
     EXPECT_EQ(1u, sender().get_err_num());
 }
 
+UCS_TEST_SKIP_COND_P(test_ucp_sockaddr, compare_cm_and_wireup_configs,
+                     no_close_protocol()) {
+    ucp_worker_cfg_index_t cm_ep_cfg_index, wireup_ep_cfg_index;
+    ucp_ep_config_key_t *cm_ep_cfg_key, *wireup_ep_cfg_key;
+
+    /* get configuration index for EP created through CM */
+    listen_and_communicate(false, SEND_DIRECTION_C2S);
+    /* TODO: remove the SKIP below and include for <uct/tcp/tcp_sockcm_ep.h>
+     *       header file, when CONNECT_TO_EP support is added for TCP */
+    if (sender().ep()->uct_eps[ucp_ep_get_cm_lane(sender().ep())]
+        ->iface->ops.ep_disconnect == uct_tcp_sockcm_ep_disconnect) {
+        UCS_TEST_SKIP_R("don't test TCP SOCKCM");
+    }
+    cm_ep_cfg_index = sender().ep()->cfg_index;
+    cm_ep_cfg_key   = &ucp_ep_config(sender().ep())->key;
+    EXPECT_NE(UCP_NULL_LANE, ucp_ep_get_cm_lane(sender().ep()));
+    disconnect(sender());
+    disconnect(receiver());
+
+    /* get configuration index for EP created through WIREUP */
+    sender().connect(&receiver(), get_ep_params());
+    ucp_ep_params_t params = get_ep_params();
+    params.field_mask     |= UCP_EP_PARAM_FIELD_USER_DATA;
+    params.user_data       = &receiver();
+    receiver().connect(&sender(), params);
+    send_recv(sender(), receiver(), send_recv_type(), 0, cb_type());
+    wireup_ep_cfg_index = sender().ep()->cfg_index;
+    wireup_ep_cfg_key   = &ucp_ep_config(sender().ep())->key;
+    EXPECT_EQ(UCP_NULL_LANE, ucp_ep_get_cm_lane(sender().ep()));
+
+    /* EP config indexes mustn't be equal */
```

Comment:
/* EP config indexes must be different because one has CM lane and the other doesn't */

Reply:
done